test/unit/pusher_test.rb

```diff
@@ -301,21 +300,12 @@ class PusherTest < ActiveSupport::TestCase
         assert_not_nil @rubygem.versions.last.info_checksum
       end
 
-      should "expire API memcached" do
-        assert_received(Rails.cache, :delete) { |cache| cache.with("info/#{@rubygem.name}") }
-        assert_received(Rails.cache, :delete) { |cache| cache.with("deps/v1/#{@rubygem.name}") }
-        assert_received(Rails.cache, :delete) { |cache| cache.with("names") }
-      end
-
-      should "purge cdn cache" do
-        Delayed::Worker.new.work_off
-        assert_received(Fastly, :purge) { |path| path.with("info/#{@rubygem.name}") }
-        assert_received(Fastly, :purge) { |path| path.with("versions") }
-        assert_received(Fastly, :purge) { |path| path.with("names") }
+      should "call GemCachePurger" do
+        assert_received(GemCachePurger, :call) { |obj| obj.with(@rubygem.name).once }
       end
 
       should "enque job for updating ES index, spec index and purging cdn" do
-        assert_difference 'Delayed::Job.count', 3 do
+        assert_difference 'Delayed::Job.count', 2 do
```

Comment:
Can someone please explain why does `Delayed::Job.count` reduces here? `Fastly.delay.purge_api_cdn(rubygem.name)` was removed and `GemCachePurger.call(rubygem.name)` (with 3 delayed jobs)  was added. Shouldn't it go up by 2?